rclpy/rclpy/action/client.py

```diff
@@ -354,12 +361,14 @@ async def execute(self, taken_data):
 
     def get_num_entities(self):
         """Return number of each type of entity used in the wait set."""
-        num_entities = self._client_handle.get_num_entities()
+        with self._lock:
+            num_entities = self._client_handle.get_num_entities()
         return NumberOfEntities(*num_entities)
 
     def add_to_wait_set(self, wait_set):
         """Add entities to wait set."""
-        self._client_handle.add_to_waitset(wait_set)
+        with self._lock:
```

Comment:
This two lines causes in my application error:
   self._client_handle.add_to_waitset(wait_set)
rclpy._rclpy_pybind11.RCLError: Failed to add 'rcl_action_client_t' to wait set: clients set is full, at  ./src/rcl/wait.c:493

I use ROS Rolling on Ubuntu 22.04 (installed from system packages).
When I commented out this two lines everything works fine.

Reply:
> I use ROS Rolling on Ubuntu 22.04 (installed from system packages).
> When I commented out this two lines everything works fine.

rolling is now supported on 24.04, https://docs.ros.org/en/rolling/Installation.html#building-from-source. can you try with ubuntu 24.04 to see if the problem still happens to you?

Reply:
It's not that easy because I use moveit2.
I'm having problems compiling moveit2 on Ubuntu 24.04.
I think it will take me some time, but I will do it.

Reply:
> rolling is now supported on 24.04, https://docs.ros.org/en/rolling/Installation.html#building-from-source. can you try with ubuntu 24.04 to see if the problem still happens to you?

It's not that easy because I use moveit2.
I'm having problems compiling moveit2 on Ubuntu 24.04.
I think it will take me some time, but I will do it.